### ISSUE TITLE
Problem with switching between append-only 1 and 0

### ISSUE THREAD

**Issue by jayjensen** [borgbackup/borg#4702]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

Question

## System information. For client/server mode post info for both machines.

Not relevant

#### Your borg version (borg -V).

borg 1.1.5

#### Operating system (distribution) and version.

Server: Ubuntu 18.04.2 LTS

#### Hardware / network configuration, and filesystems used.

Filesystem: XFS

#### How much data is handled by borg?

To handle: About 100 GB
See question.

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

borg prune ...

## Describe the problem you're observing.

**Introduction:**
The repository itself is configured with append-only=0. 
However, the client may only save in append-only=1, which is forced by the SSH-Server. So no actual deletions are made when the client prunes, as expected.
From time to time, a pruning is performed directly on the server to actually clean up the repository.

**Problem:**
The size of the backup data is about 100GB.
After the serverside (append-only=0) pruning, the size of the repopath/data directory before and after the pruning is about 780GB.
So no data gets actually deleted.

```
                       Original size      Compressed size    Deduplicated size
All archives:               91.01 GB             91.36 GB             91.36 GB

                       Unique chunks         Total chunks
Chunk index:                   35846                35846
```

**Question:**
Maybe I'm getting it wrong, but shouldn't the data marked as deleted get deleted, when switching to append-only 0 and pruning ?

Thanks in advance,

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

This problem occurs repeatedly. Until now, to fix this the repo was re-created.

#### Include any warning/errors/backtraces from the system logs

The log doesn't show errors.

**Comment by ThomasWaldmann** [comment]
What's the log output of the server side prune command?
Please run it like `borg prune -v ...`.

Also, there are quite more recent borg releases. I don't remember fixing any bug related to this, but in general it might be a good idea to use something more recent. You can get the latest one from the maintainer's ppa.

**Comment by jayjensen** [comment]
I think I have to apologize for the question.

Normally the pruning is script driven.
However, because of the amount of data and the diversity of the data, pruning for exactly this repository takes quite a while. 
Apparently the SSH connection was interrupted regularly and the pruning process was not finished completely.
Unfortunately, this was not visible in the log file.

**What I did now:**
Taking into account your previous answer, I have now performed the operation non-script controlled.
The interactive pruning and forced keeping the SSH connection open has now led to a successful cleanup of the repository.

So, excuse me for asking and thank you very much.
